case_studies/dc2_cataloging/utils/convnet.py

```diff
@@ -0,0 +1,162 @@
+# adapted from the convnet.py in commit 53936977c3a87f8ba937d75bec0fd951eb5c23d9
+
+import torch
+from torch import nn
+
+# The code in this file is based on the following repository:
+# https://github.com/ultralytics/yolov5/
+
+
+class ConvBlock(nn.Module):
+    def __init__(
+        self, in_channels, out_channels, kernel_size=3, stride=1, padding=1, use_group_norm=False
+    ):
+        super().__init__()
+        self.conv = nn.Conv2d(in_channels, out_channels, kernel_size, stride, padding, bias=False)
+        self.norm = (
+            nn.BatchNorm2d(out_channels, eps=0.001, momentum=0.03)
+            if not use_group_norm
+            else nn.GroupNorm(out_channels // 16, out_channels)
+        )
+        self.activation = nn.SiLU(inplace=True)
+
+    def forward(self, x):
+        return self.activation(self.norm(self.conv(x)))
+
+
+class Detect(nn.Module):
+    def __init__(self, in_channels, out_channels):
+        super().__init__()
+        self.conv = nn.Conv2d(in_channels, out_channels, 1, 1, 0, bias=True)
+
+    def forward(self, x):
+        return self.conv(x).permute([0, 2, 3, 1])
+
+
+class Bottleneck(nn.Module):
```

Comment:
Better not to duplicate code. Please use the versions of these in `bliss.encoder.convnet` if possible. It's only if you're modify them that we'd want to include them here.